### DISCUSSION TITLE
Possibility of getting a signed firefox version?

### DISCUSSION THREAD

**Discussion by Zachruff** [younesaassila/ttv-lol-pro#190] [Q&A] [answered]
Since the firefox version is unsigned and firefox is stupid from what I can tell I need to be using another build so I got the Extended Support Release as I dont want to have two different installations at once (afaik developer edition counts as its own thing when installed) but being on ESR causes problems for other things I am using firefox for. Is there any chance of getting the extension signed so I can just install it on a vanilla firefox instance permanently? Thank you

**Comment by younesaassila** [comment]
I've published the extension for review on Firefox's Add-ons Store. Unfortunately, not only are they super slow (took 10 days to respond to v2.0.0), they've rejected my 2.0.0 build for "remote code execution is not allowed" because they couldn't be bothered to take a close look at the code and to realize the script I'm injecting into Twitch is bundled with the extension (i.e. they can read its source code).
I've added a comment above the line in question for the reviewers and published v2.0.1 for review 2 days ago. We'll see what they say in 8 days...

**Reply by younesaassila** [reply, marked as answer]
Mozilla has finally approved the extension! https://addons.mozilla.org/addon/ttv-lol-pro/

**Comment by sn-o-w** [comment]
@Zachruff You can patch Firefox with this: https://github.com/xiaoxiaoflood/firefox-scripts#readme

After you installed userChromeJS, you need to go to `about:addons` and just manually install the Firefox zip file you download from [here](https://github.com/younesaassila/ttv-lol-pro/releases/latest).

I tested it a few days ago, installing TTV LOL PRO on Firefox 114. It works like a charm.

**Comment by sn-o-w** [comment]
**LATER EDIT**: I've simplified that thing to only allow installing unsigned add-ons, nothing more than that. 😊

So you can ignore my previous comment and proceed to this:

[install-unsigned-addons.zip](https://github.com/younesaassila/ttv-lol-pro/files/11780247/install-unsigned-addons.zip)

- Close Firefox. 
- Download the above zip archive, extract its content to `C:\Program Files\Mozilla Firefox`
- Open Firefox, go to `about:addons#extensions` and manually install any unsigned addon you want.

**Reply by sn-o-w** [reply]
**UPDATE**: This should work on Firefox v115+.

[install-unsigned-addons-v2.zip](https://github.com/younesaassila/ttv-lol-pro/files/12440891/install-unsigned-addons-v2.zip)

Same installation steps.

**Comment by PoorPocketsMcNewHold** [comment]
At least, could you potentially release the signed .xpi files eitherway? It installs fine for other project refused from the Mozilla add-on store, such as https://github.com/FastForwardTeam/FastForward or https://github.com/iamadamdev/bypass-paywalls-chrome on Firefox v113.0.1.

**Reply by younesaassila** [reply]
Signing extensions is also done via the Mozilla Add-on Store.
To sign the v1.9.x updates (distributed via GitHub), I also had to wait 7-10 days for Mozilla to approve them.

**Reply by younesaassila** [reply]
Incredible timing, Mozilla has finally approved the extension! https://addons.mozilla.org/addon/ttv-lol-pro/
